integration-tests/src/test/java/com/datastax/oss/driver/core/cql/PreparedStatementIT.java

```diff
@@ -444,9 +447,32 @@ public void should_fail_fast_if_id_changes_on_reprepare() {
       executeDdl("DROP TABLE prepared_statement_test");
       executeDdl("CREATE TABLE prepared_statement_test (a int PRIMARY KEY, b int, c int)");
 
-      assertThatThrownBy(() -> session.execute(preparedStatement.bind(1)))
-          .isInstanceOf(IllegalStateException.class)
-          .hasMessageContaining("ID mismatch while trying to reprepare");
+      // JAVA-3065: determine if cassandra/dse version has CASSANDRA-15252 fix
+      boolean isCassandra15252Fixed = false;
+      if (ccmRule.getDseVersion().isPresent()) {
+        // no DSE versions have recent enough cassandra yet
+        isCassandra15252Fixed = false;
+      } else {
+        // cassandra must be 3.0.26+, 3.11.12+ or 4.0.2+
+        Version cassandraVersion = ccmRule.getCassandraVersion();
+        if (cassandraVersion.compareTo(Version.parse("4.0.0")) >= 0) {
+          isCassandra15252Fixed = cassandraVersion.compareTo(Version.parse("4.0.2")) >= 0;
+        } else if (cassandraVersion.compareTo(Version.parse("3.11.0")) >= 0) {
+          isCassandra15252Fixed = cassandraVersion.compareTo(Version.parse("3.11.12")) >= 0;
+        } else if (cassandraVersion.compareTo(Version.parse("3.0.0")) >= 0) {
+          isCassandra15252Fixed = cassandraVersion.compareTo(Version.parse("3.0.26")) >= 0;
+        }
+      }
+
+      ThrowableAssert.ThrowingCallable errorTrigger =
+          () -> session.execute(preparedStatement.bind(1));
+      if (isCassandra15252Fixed) {
+        assertThatCode(errorTrigger).doesNotThrowAnyException();
+      } else {
+        assertThatThrownBy(errorTrigger)
+            .isInstanceOf(IllegalStateException.class)
+            .hasMessageContaining("ID mismatch while trying to reprepare");
+      }
```

Comment:
Within the integration test framework this kind of thing is normally handled via the [CassandraRequirement annotation](https://github.com/datastax/java-driver/blob/4.x/test-infra/src/main/java/com/datastax/oss/driver/api/testinfra/CassandraRequirement.java).  You can find examples in [other test methods of PreparedStatementIT](https://github.com/datastax/java-driver/blob/4.x/integration-tests/src/test/java/com/datastax/oss/driver/core/cql/PreparedStatementIT.java#L154) (as well as other integration tests).  Enforcement of these annotations can be found in [BaseCcmRule](https://github.com/datastax/java-driver/blob/4.x/test-infra/src/main/java/com/datastax/oss/driver/api/testinfra/ccm/BaseCcmRule.java#L76-L132).

I'd very strongly prefer not to introduce multiple methods for supporting this kind of behaviour unless it's absolutely necessary, but it's worth pointing out that the current impl does have a limitation of relevance for us here.  To really do this right you'd want to be able to (a) attach multiple annotations to a given test method, one for each major.minor version, with a min or max version specified as appropriate and (b) have BaseCcmRule evaluate _all_ annotation on the test method and enforce all of them.  After a quick review of the code I don't think that change is actually that bad, so maybe this is an opportunity to add a new feature (support for multiple annotations) in order to avoid having to introduce another way to handle version constraints.

Reply:
Added new annotation: `BackendRequirement` which supports annotation repetition and am now using this with `should_fail_fast_if_id_changes_on_reprepare`. I've left the previous anotations in for now because I think there is some more tidying up, e.g. for tests which have been copied to support multiple c*/dse requirement, I've filed JAVA-3069 to track this work.

Reply:
Ignored error in intellij test output looks like:
```org.junit.AssumptionViolatedException: Test requires one of:
  - C* 3.0.26 or greater, but less than 3.11.0
  - C* 3.11.12 or greater, but less than 4.0.0
  - C* 4.0.2 or greater
but configuration is C* 4.0.0.```